### ISSUE TITLE
Sender address not shown on latest firmware 1.2.5

### ISSUE THREAD

**Issue by mxmauro** [algorand/ledger-app-algorand#26]
Hi,

When sending a TX the sender's address is not shown.

Although this can appear to be a bit silly, most people uses it to check/double-check funds are being sent from the correct account. mainly, if the user owns more than one device configured with different addresses.

This also can present a security issue if the UI requesting signing is not user friendly or if they UI does not show the correct sender address (by mistake or on purpose).

Regards,
Mauro.

**Comment by pbennett** [comment]
What other Ledger apps have you seen do this?  The ETH Ledger app for example doesn't do this.

**Comment by mxmauro** [comment]
Hi @pbennett I don't know about ETH app but seems people in Algorand world does like that feature. We have lots of people writing us (myalgo.com) asking to get that functionality back.

**Comment by patrickbsf** [comment]
BTC, ETH ledger apps, it's all about confirming the recipient address.  The sender is typically self-evident (or meaningless in case of BTC because of UTXOs).

**Comment by mxmauro** [comment]
Checked BTC, ETH, and Algorand apps. The behavior of the three are similar.

All receives an accountId or BIP32, Algorand's one can default to the accountId zero. But all relies on the app sending the TX to sign, to send the correct sender.

If the user selects the wrong account. if the app has a bug and build the TX with the wrong account or, if a malicious, hooked or hacked app on purpose builds the TX with the wrong account, the user is unable to know and verify it on the device.

Basically I'm forwarding user's feedback we receive thru our wallet.

**Comment by pbennett** [comment]
Keep in mind, the Algorand app doesn't support multiple accounts at the moment - at all.  You can't send from multiple accounts.  It only supports one. (I'd love more attention to my issue re. that #22 )

Even if it did support multiple accounts, I guess I just don't see a strong attack vector.  Malicious actors will often manipulate the receiving address (modifying your clipboard, or modifying a QR code, etc.) so that they receive the funds, and not your intended recipient.  So, validating the receiving address is obviously critical.
Changing the send account wouldn't provide a way for them to receive money.
I think the Ledger created apps kind of set the expectation of app behaviors, and everyone is going to pattern themselves after that.  Considering it hasn't happened already in all the other apps makes me think most people don't consider it an issue?  It wouldn't be the worst thing to include - but obviously the Algorand app has to support multiple accounts first ;)

**Comment by mxmauro** [comment]
Sorry for the late response @pbennett . Basically all is resumed to people using the app feels confortable and safer seeing the sender address as part of the flow and, personally I think it is a nice feature other apps does not have.

**Comment by tsachiherman** [comment]
I have created two templates - one for bug reports, and one for feature requests. Could you please recreate this issue using the new issue template and close this one ?
